hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServer.java

```diff
@@ -172,10 +306,10 @@ private ByteBufAllocator getChannelAllocator(Configuration conf) throws IOExcept
     }
   }
 
-  // will be overriden in tests
+  // will be overridden in tests
   @InterfaceAudience.Private
-  protected NettyRpcServerPreambleHandler createNettyRpcServerPreambleHandler() {
-    return new NettyRpcServerPreambleHandler(NettyRpcServer.this);
+  protected NettyServerRpcConnection createNettyServerRpcConnection(Channel channel) {
```

Comment:
this change is no longer strictly necessary, but i kept it because i think it's a cleaner way to do what we want to do. in all cases, tests were overriding `createNettyRpcServerPreambleHandler()` just so they could inject a special `NettyServerRpcConnection`. So it works better to simply override the creation of the NettyServerRpcConnection directly.